### PR TITLE
deps: Update golang version and dependencies

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
These changes are in this PR:
- Updated golang version to `1.22.4`.
- Created golang workspace, because this is a multi-module repository.
- Removed unused indirect dependencies in the `api` module.
- Updated various dependencies.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -178,7 +178,7 @@ vet:
 .PHONY: vendor
 vendor:
 	go mod tidy
```
Comment (0xFelix) [inline Makefile:180]: Will this tidy both modules?

File: internal/operands/metrics/resources.go [hunk 2]
```diff
@@ -82,7 +83,7 @@ func newServiceMonitorCR(namespace string) *promv1.ServiceMonitor {
 					Scheme: "https",
 					TLSConfig: &promv1.TLSConfig{
 						SafeTLSConfig: promv1.SafeTLSConfig{
-							InsecureSkipVerify: true,
```
Comment (0xFelix) [inline internal/operands/metrics/resources.go:85]: It's ok to skip TLS?

File: Makefile (hunk 1 above)
Comment (akrejcir) [reply Makefile:181]: fixed

File: internal/operands/metrics/resources.go (hunk 2 above)
Comment (akrejcir) [reply internal/operands/metrics/resources.go:85]: I'm not sure, this is old code. I will open an issue to investigate.

Comment (akrejcir) [reply internal/operands/metrics/resources.go:85]: https://github.com/kubevirt/ssp-operator/issues/1009

File: Dockerfile [hunk 3]
```diff
@@ -3,7 +3,7 @@ FROM registry.access.redhat.com/ubi9/ubi-minimal as builder
 
 RUN microdnf install -y make tar gzip which && microdnf clean all
 
-RUN curl -L https://go.dev/dl/go1.21.6.linux-amd64.tar.gz | tar -C /usr/local -xzf -
+RUN curl -L https://go.dev/dl/go1.22.4.linux-amd64.tar.gz | tar -C /usr/local -xzf -
```
Comment (0xFelix) [inline Dockerfile:6]: Same question as in https://github.com/kubevirt/ssp-operator/pull/1010, use the latest dot release?

**Comment by jcanocan** [comment]
Nice job @akrejcir!
/lgtm

File: api/go.mod [hunk 4]
```diff
@@ -1,30 +1,30 @@
 module kubevirt.io/ssp-operator/api
 
-go 1.21
+go 1.22.5
```
Comment (dominikholler) [inline api/go.mod:3]: Is there any chance to avoid the z part of the version number?

**Review by 0xFelix — COMMENTED** [review]
/approve

**Pull request merged**